pre_commit/metrics.py

```diff
@@ -0,0 +1,64 @@
+from contextlib import contextmanager
+from dataclasses import dataclass
+from datetime import datetime, timezone
+import shlex
+import subprocess
+from typing import List, Generator, Dict, Optional
+import json
+
+@dataclass
```

Comment:
Brownie points: You could set this as a frozen class so that once the attrs are set, they can't (reasonably) be changed.
`@dataclass(frozen=True)`